#!/usr/bin/env python
PACKAGE = "fabric_grasping"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# viewer parameters
gen.add("int_displayed_normals", int_t, 0, "Sets the number of normals displayed", 50, 1, 200)
gen.add("double_length_normals", double_t, 0, "Sets the length of normals displayed", 0.005, 0.001, 0.02)

# listener parameters
gen.add("int_setKSearch", int_t, 0, "Set number of neighbors included into calculating normals", 30, 3, 300)
gen.add("double_setMinRadius", double_t, 0, "Set minimum RANSAC cylinder radius", 0.002, 0.0005, 0.0035)
gen.add("double_setMaxRadius", double_t, 0, "Set maximum RANSAC cylinder radius", 0.012, 0.0036, 0.15)
gen.add("double_setNormalDistanceWeight", double_t, 0, "Set ransac parameter: normal distance weight", 0.01, 0, 1)
gen.add("double_setDistanceThreshold", double_t, 0, "Set ransac parameter: distance threshold", 0.003, 0.001, 0.01)
gen.add("string_frame", str_t, 0, "Set frame", "mount_plate_link")
gen.add("bool_crop_box", bool_t, 0, "Enable cropbox", True)

gpd_selector_enum = gen.enum([gen.const("overall_z_max", int_t, 0, "Returns inlier Z-Max grasping point"),
                        gen.const("overall_z_max_fixed_orientation", int_t, 1, "Returns inlier Z-Max grasping point with fixed orientation"),
                        gen.const("overall_centroid", int_t, 2, "Returns inlier Z-Max grasping point"),
                        gen.const("cylinder_z_max", int_t, 3, "Returns inlier Z-Max grasping point"),
                        gen.const("cylinder_centroid", int_t, 4, "Returns inlier Z-Max grasping point")],
                        "Select grasping point method")

gen.add("gp_method", int_t, 0, "Select grasping point method", 0, 0, 4, edit_method=gpd_selector_enum)

exit(gen.generate(PACKAGE, "fabric_grasping", "parameters"))